File: Go/potencia.go
package main

import (
	"fmt"
)

func main() {
	var base, exponente int
	fmt.Print("Insira a base: ")
	fmt.Scan(&base)
	fmt.Print("Insira o expoente: ")
	fmt.Scan(&exponente)
	if exponente < 0 {
		fmt.Println("O expoente deve ser um número inteiro não negativo.")
		return
	}
	resultado := base
	for i := 1; i < exponente; i++ {
		resultado *= base
	}
	fmt.Printf("%d elevado a %d é igual a %d\n", base, exponente, resultado)
}

File: haskell/potencia.hs
potencia :: Int -> Int -> Int
potencia x y = x^y

main :: IO ()
main = do
    putStrLn "Insira a base:"
    base <- getLine
    putStrLn "Insira o expoente:"
    expoente <- getLine
    let baseInt = read base :: Int
    let expoenteInt = read expoente :: Int
    let resultado = potencia baseInt expoenteInt
    putStrLn ("O resultado de " ++ show baseInt ++ " elevado a " ++ show expoenteInt ++ " eh " ++ show resultado)